targets/csharp/make.js

```diff
@@ -262,36 +265,63 @@ function getAuthParams(apiCall, isApiInstance = false) {
     if (apiCall.auth === "EntityToken")
-        return tabbing + "if (PlayFabSettings.EntityToken == null) throw new PlayFabException(PlayFabExceptionCode.EntityTokenNotSet, \"Must call GetEntityToken before calling this method\");\n";
-    if (apiCall.auth === "SessionTicket")
-        return tabbing + "if (PlayFabSettings.ClientSessionTicket == null) throw new PlayFabException(PlayFabExceptionCode.NotLoggedIn, \"Must be logged in to call this method\");\n";
+        return tabbing + "#if !DISABLE_PLAYFABENTITY_API\n"
+	        + tabbing + "    if ((request.AuthenticationContext?.EntityToken ?? PlayFabSettings.EntityToken) == null) throw new PlayFabException(PlayFabExceptionCode.EntityTokenNotSet, \"Must call GetEntityToken before calling this method\");\n"
+			+ tabbing + "#else\n"    
+			+ tabbing + "    if ((PlayFabSettings.EntityToken) == null) throw new PlayFabException(PlayFabExceptionCode.EntityTokenNotSet, \"Must call GetEntityToken before calling this method\");\n"
+			+ tabbing + "#endif\n";
+	if (apiCall.auth === "SessionTicket")
+        return tabbing + "#if !DISABLE_PLAYFABCLIENT_API\n"
+	        + tabbing + "    var clientSessionTicket = request.AuthenticationContext?.ClientSessionTicket ?? PlayFabSettings.ClientSessionTicket;\n"
+            + tabbing + "    if (clientSessionTicket == null) throw new PlayFabException(PlayFabExceptionCode.NotLoggedIn, \"Must be logged in to call this method\");\n"
+			+ tabbing + "#else\n"    
+			+ tabbing + "    var clientSessionTicket = PlayFabSettings.ClientSessionTicket;\n"
+			+ tabbing + "#endif\n";
     if (apiCall.auth === "SecretKey" && isApiInstance === false)
         return tabbing + "if (PlayFabSettings.DeveloperSecretKey == null) throw new PlayFabException(PlayFabExceptionCode.DeveloperKeyNotSet, \"Must have PlayFabSettings.DeveloperSecretKey set to call this method\");\n";
     if (apiCall.auth === "SecretKey" && isApiInstance === true)
-        return tabbing + "var developerSecretKey = request.AuthenticationContext?.DeveloperSecretKey ?? (authenticationContext?.DeveloperSecretKey ?? PlayFabSettings.DeveloperSecretKey);\n            if (developerSecretKey == null) throw new PlayFabException(PlayFabExceptionCode.DeveloperKeyNotSet, \"DeveloperSecretKey is not found in Request, Server Instance or PlayFabSettings\");\n";
+        return tabbing + "#if ENABLE_PLAYFABSERVER_API || ENABLE_PLAYFABADMIN_API\n"
+			+ tabbing + "    var developerSecretKey = request.AuthenticationContext?.DeveloperSecretKey ?? (authenticationContext?.DeveloperSecretKey ?? PlayFabSettings.DeveloperSecretKey);\n"
+			+ tabbing + "    if (developerSecretKey == null) throw new PlayFabException(PlayFabExceptionCode.DeveloperKeyNotSet, \"DeveloperSecretKey is not found in Request, Server Instance or PlayFabSettings\");\n"
+			+ tabbing + "#else\n"    
+			+ tabbing + "    var developerSecretKey = PlayFabSettings.DeveloperSecretKey;\n"
+			+ tabbing + "#endif\n";
```

Comment:
Theoretically, it should not be required to check ENABLE_PLAYFABSERVER_API  on every server-authenticated API method, because the entire API file should be wrapped properly.
If you're experiencing a problem that requires this, you should step back and try and figure out why the entire file isn't wrapped in ENABLE_PLAYFABSERVER_API.

Reply:
as we talked in #31's PR I don't think there is an error because we already generate projects with these macros but to prevent a future confision I add them. Should I remove?

![image](https://user-images.githubusercontent.com/4026209/53132071-96e1a600-357f-11e9-8256-585630055057.png)